src/components/UserList/index.js

```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import './style.css'
+
+function UserList(props) {
+  return (
+    <ul>
+      {props.users.map((name, index) => <li onClick={props.handleNameClick} key={index}>{props.users[index].name}</li>)}
```

Comment:
This is the way I would pass the index of the users to the handleNameClick function
```suggestion
      {props.users.map((name, index) => <li onClick={() => props.handleNameClick(index)} key={index}>{props.users[index].name}</li>)}
```